src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -1288,6 +1290,22 @@ object HighwayTools : Module() {
             }
         }
 
+        /* Checks if either a boat or a minecart is blocking the highway, and if so punches it */
+        fun punchBlocking(): Boolean {
+            for (entity in mc.world.loadedEntityList) {
+                if ((entity is EntityMinecart ||
+                    entity is EntityBoat)
+                    && getDistance(mc.player.positionVector, entity.positionVector) <= maxReach.value) {
+                    val rotation = RotationUtils.getRotationTo(entity.positionVector, true)
+                    setRotation(rotation)
+                    mc.playerController.attackEntity(mc.player, entity)
```

Comment:
Would u change it to the Hotbar where u got a sword and the fact it needs to be punched more than one, cus if there are a lot of boats and lag and no sword it can take forever and maybe get the player robber banding and disconnect?

Reply:
I will add a check for swords. Thanks for the suggestion